docs/proposals/scheduling/policy-preemption/README.md

```diff
@@ -0,0 +1,214 @@
+kind: PropagationPolicy
+metadata:
+  name: propagation-labelselectormatch
+spec:
+  resourceSelectors:
+    - apiVersion: apps/v1
+      kind: Deployment
+      labelSelector:
+        matchLabels:
+          app: nginx
+  placement:
+    clusterAffinity:
+      clusterNames:
+        - member2
+---
+# namematch.yaml
+apiVersion: policy.karmada.io/v1alpha1
+kind: PropagationPolicy
+metadata:
+  name: propagation-namematch
+spec:
+  resourceSelectors:
+    - apiVersion: apps/v1
+      kind: Deployment
+      name: nginx
+  placement:
+    clusterAffinity:
+      clusterNames:
+        - member3
+```
+
+## Design Details
+
+### API change
+
+#### PropagationPolicy API change
+
+This proposal proposes a new field `Preemption` for determining 
+whether to allow this policy to preempt other policy.
+
+```go
+type PropagationSpec struct {
+    // +kubebuilder:default=0
+    Priority *int `json:"priority,omitempty"`
+
+    // Preemption indicates whether the policy is allowed to preempt low-priority policies.
+    // If true, it will preempt those workloads whose previously matched policy has a lower priority than it.
+    // Defaults to false.
+    // +kubebuilder:default=false
+    Preemption *bool `json:"preemption,omitempty"`
+}
+```
+
+For example, assume that policy `nginx-propagation` has bound workload `nginx`:
+
+```yaml
+apiVersion: policy.karmada.io/v1alpha1
+kind: PropagationPolicy
+metadata:
+  name: nginx-propagation
+spec:
+  resourceSelectors:
+    - apiVersion: apps/v1
+      kind: Deployment
+      name: nginx
+  priority: 2
+  placement:
+    clusterAffinity:
+      clusterNames:
+        - member1
+        - member2
+```
+
+```yaml
+apiVersion: work.karmada.io/v1alpha2
+kind: ResourceBinding
+metadata:
+  labels:
+    propagationpolicy.karmada.io/name: nginx-propagation
+    propagationpolicy.karmada.io/namespace: default
+  name: nginx-deployment
+  namespace: default 
+```
+
+Assume that there is a high-priority policy which allows preepmtion:
+
+```yaml
+apiVersion: policy.karmada.io/v1alpha1
+kind: PropagationPolicy
+metadata:
+  name: nginx-propagation-preemption
+spec:
+  resourceSelectors:
+    - apiVersion: apps/v1
+      kind: Deployment
+      name: nginx
+  priority: 3
+  preemption: true
+  placement:
+    clusterAffinity:
+      clusterNames:
+        - member3
+```
+
+`nginx-propagation-preemption` will preempt `nginx-propagation` and the deployment will be scheduled to `member3`.
+
+### Components change
+
+#### karmada-controller-manager
+
+Currently, when a new policy creates/updates, `detector` will first check whether existing objects are no longer matched by the current policy to handle the changes of `resourceSelectors`. If yes, clean the labels on the unmatched object. Then it will find the corresponding ResourceBinding and make sure that PropagationPolicy's updates
+can be synchronized to ResourceBinding. Finally, it will try to find the matched resource templates which have not been scheduled.
```

Comment:
Are you trying to explain the `current` behavior? Just feel some misleading.

Reply:
Yes, sorry, maybe the description is not accurate.